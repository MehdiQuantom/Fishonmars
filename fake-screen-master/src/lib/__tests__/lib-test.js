import {
  sleep,
  genID,
  isStr,
  twoDigits,
  getDayStr,
  getMonthStr,
} from 'src/lib';

it('sleep lib', async () => {
  const now = new Date();
  await sleep(1000);
  expect(new Date() - now >= 1000);
});

it('genID', () => {
  const getID = genID();
  expect([getID(), getID(), getID()]).toEqual([0, 1, 2]);
});

it('isPath', () => {
  const isA = isStr('A');
  expect(isA('A')).toBeTrue();
  expect(isStr('A')('B')).toBeFalse();
});

it('fix number to two digits', () => {
  expect(twoDigits(1)).toBe('01');
  expect(twoDigits(10)).toBe('10');
  expect(twoDigits(-1)).toBe('-1');
});

it('get correct day string', () => {
  expect(getDayStr(1)).toBe('Monday');
  expect(getDayStr(8)).toThrowErrorMatchingSnapshot();
});

it('get correct Month string', () => {
  expect(getMonthStr(1)).toBe('February');
  expect(getMonthStr(-1)).toThrowErrorMatchingSnapshot();
});
