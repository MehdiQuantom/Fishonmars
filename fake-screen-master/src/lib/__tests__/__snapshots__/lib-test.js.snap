// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get correct Month string 1`] = `"received is not a function"`;

exports[`get correct day string 1`] = `"received is not a function"`;
